### DISCUSSION TITLE
When opening save file, check if object is in use.

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#20698] [Feature requests]
Sometimes, I enable objects in a park, and then I do not use them. If I clean my objects folder, the "official" object is not exported. However, because it is enabled in object selection it causes an error on opening the file. It would be better if when opening a save, the game checks if the object is merely selected, or if it is present in the park. I have attached a save file with this issue.
[73.zip](https://github.com/OpenRCT2/OpenRCT2/files/12306800/73.zip)
![asddasdasdas](https://github.com/OpenRCT2/OpenRCT2/assets/94884086/c3f2f9ac-8340-42b2-ba2a-95089a62ee2e)
I believe this object was listed as "official" by mistake

**Comment by ocalhoun6** [comment]
It should probably still give an error if the object is selected but unused -- but the error should be dismissable, allowing you to deselect the object and load the park anyway.

It might be important to still show an error with continue/cancel buttons, because if you've carefully curated the object selection on a park, you might _not_ want those objects to be deselected if the game can't (currently) find them. And, at any rate, it could be helpful to know that there's a problem, rather than ending up thinking, "Hey, where'd my selected objects go?" when you open the file.

**Comment by duncanspumpkin** [comment]
This would be hard to do as it requires modifying the save without loading it. At present we don't have that capability. Why don't you just remove your unused objects from your save using the `remove_unused_objects` command before you delete stuff from your object folder.

**Reply by spacek531** [reply]
Sometimes, I don't plan ahead.

**Reply by 733737** [reply]
The intention of this discussion, duncanspumpkin, is to allow for the implementation within years. It is a long term goal of mine to boost the potentiality of object nondissonance.

**Reply by ZehMatt** [reply]
Scanning for unused objects in the park would also increase the load times of parks, there are also legitimate cases where unused objects are desired to be included, like parks that are not finished yet and more than one person works on it. As for official objects they are not exported for a reason as we expect them to exist for every standard installation, if you delete them then its an user error.

**Reply by 733737** [reply]
I agree with your statement. This is why it would be, hypothetically mind you, an _optional_ hex edit hack. If you think this is off the table, I am willing to close this discussion.
